roles/iperf3/tasks/main.yml

```diff
@@ -1,75 +1,43 @@
 ---
-- name: Get current state
-  k8s_facts:
-    api_version: ripsaw.cloudbulldozer.io/v1alpha1
-    kind: Benchmark
-    name: '{{ ansible_operator_meta.name }}'
-    namespace: '{{ operator_namespace }}'
-  register: resource_state
-
-- operator_sdk.util.k8s_status:
-    api_version: ripsaw.cloudbulldozer.io/v1alpha1
-    kind: Benchmark
-    name: "{{ ansible_operator_meta.name }}"
-    namespace: "{{ operator_namespace }}"
-    status:
-      state: Starting
-      complete: false
-  when: resource_state.resources[0].status.state is not defined
-
-- name: Update current state
-  k8s_facts:
-    api_version: ripsaw.cloudbulldozer.io/v1alpha1
-    kind: Benchmark
-    name: '{{ ansible_operator_meta.name }}'
-    namespace: '{{ operator_namespace }}'
-  register: resource_state
-
 - block:
 
   - name: Start Server(s)
     k8s:
       definition: "{{ lookup('template', 'server.yml.j2') | from_yaml }}"
     with_sequence: start=1 count={{ workload_args.pairs|default('1')|int }}
 
-  - name: Update state to Starting Server
-    operator_sdk.util.k8s_status:
-      api_version: ripsaw.cloudbulldozer.io/v1alpha1
-      kind: Benchmark
-      name: "{{ ansible_operator_meta.name }}"
-      namespace: "{{ operator_namespace }}"
-      status:
-        state: "Starting Server"
+  - include_role:
+      name: benchmark_state
+      tasks_from: set_state
+    vars:
+      state: "Starting Server"
 
-  when: resource_state.resources[0].status.state == "Starting"
+  when: benchmark_state.resources[0].status.state == "Building"
 
 - block:
 
   - name: Wait for pods to be running....
-    k8s_facts:
+    k8s_info:
       kind: Pod
       api_version: v1
       namespace: '{{ operator_namespace }}'
       label_selectors:
         - app = iperf3-bench-server-{{ trunc_uuid }}
     register: server_pods
 
-  - name: Update state to Starting Clients
-    operator_sdk.util.k8s_status:
-      api_version: ripsaw.cloudbulldozer.io/v1alpha1
-      kind: Benchmark
-      name: "{{ ansible_operator_meta.name }}"
-      namespace: "{{ operator_namespace }}"
-      status:
-        state: "Starting Clients"
+  - include_role:
+      name: benchmark_state
+      tasks_from: set_state
+    vars:
+      state: "Starting Clients"
     when: "workload_args.pairs|default('1')|int == (server_pods | json_query('resources[].status.podIP')|length) and (server_pods | json_query('resources[].status.phase') is defined and workload_args.pairs|default('1')|int == server_pods | json_query('resources[].status[]')|selectattr('phase','match','Running')|list|length)"
 
-  when: resource_state.resources[0].status.state == "Starting Server"
+  when: benchmark_state.resources[0].status.state == "Starting Server"
```

Comment:
Unrelated again and minor, the other statuses are `Starting Servers` (plural) regardless of count of servers started. `iperf3` is the only one that's singular.

Reply:
You're right, I just tried to keep the current semantics here, the major changes are that all benchmarks set now a Completed or Failed state